qt-everywhere: update to 5.9.2, add qtserialport and qttools

thoradia opened the pull request:
Switching from qtbase to qt-everywhere allows to easily add Qt modules, here qttools which proves useful to build add-ons. By the way qt-everywhere was requested here: https://forum.libreelec.tv/thread/5606-qt-support-in-libreelec/. qt-everywhere requires pcre2. The purpose of the first commit is to easily track the differences between the qtbase and qt-everywhere package.mk.

Is hyperion building for LibreELEC 9.0?

CvH commented on packages/addons/addon-depends/pcre2/package.mk:
```diff
@@ -0,0 +1,51 @@
+
+PKG_NAME="pcre2"
+PKG_VERSION="10.30"
+PKG_LICENSE="OSS"
```
`PKG_LICENSE="BSD"` :)

CvH commented on packages/addons/addon-depends/pcre2/package.mk:
```diff
@@ -0,0 +1,51 @@
+PKG_LICENSE="OSS"
+PKG_SITE="http://www.pcre.org/"
+PKG_URL="http://ftp.csx.cam.ac.uk/pub/software/programming/pcre/$PKG_NAME-$PKG_VERSION.tar.bz2"
+PKG_SHA256="90bd41c605d30e3745771eb81928d779f158081a51b2f314bbcc1f73de5773db"
```
pls move this line under PKG_VERSION

CvH commented on packages/addons/addon-depends/pcre2/package.mk:
```diff
@@ -0,0 +1,51 @@
+PKG_VERSION="10.30"
+PKG_LICENSE="OSS"
+PKG_SITE="http://www.pcre.org/"
+PKG_URL="http://ftp.csx.cam.ac.uk/pub/software/programming/pcre/$PKG_NAME-$PKG_VERSION.tar.bz2"
```
`PKG_URL="https://ftp.pcre.org/pub/pcre/$PKG_NAME-$PKG_VERSION.tar.bz2"` sounds more reasonable ;)

CvH commented on packages/addons/addon-depends/qt-everywhere/package.mk:
```diff
@@ -16,18 +17,16 @@
-PKG_URL="http://download.qt.io/official_releases/qt/5.6/$PKG_VERSION/submodules/$PKG_NAME-opensource-src-$PKG_VERSION.tar.xz"
-PKG_DEPENDS_TARGET="pcre zlib"
+PKG_URL="http://download.qt.io/official_releases/qt/5.9/$PKG_VERSION/single/$PKG_NAME-opensource-src-$PKG_VERSION.tar.xz"
+PKG_SHA256="7b41a37d4fe5e120cdb7114862c0153f86c07abbec8db71500443d2ce0c89795"
```
pls move this line under PKG_VERSION

CvH commented on packages/addons/addon-depends/qt-everywhere/package.mk:
```diff
@@ -16,18 +17,16 @@
 PKG_SOURCE_DIR="$PKG_NAME-opensource-src-$PKG_VERSION"
-PKG_SHORTDESC="A cross-platform application and UI framework"
 PKG_LONGDESC="A cross-platform application and UI framework"
-PKG_AUTORECONF="no"
```
pls don't remove `PKG_AUTORECONF="no"`  (no intern agreements yet if we drop this or not)

thoradia commented:
Thank you @CvH 
I rebased

Raybuntu commented:
Thank you @thoradia. I have actually cherry picked qt-everywhere from you to get hyperion.ng working. One question. Why the need for pcre2? I have been using just pcre. If you need pcre in a higher version why not bump the package.

Raybuntu commented:
And can you please rebase to resolve conflicts.

MilhouseVH commented on packages/addons/addon-depends/pcre2/package.mk:
```diff
@@ -0,0 +1,51 @@
+PKG_DEPENDS_TARGET="toolchain"
+PKG_LONGDESC="The PCRE library is a set of functions that implement regular expression pattern matching using the same syntax and semantics as Perl 5. PCRE has its own native API, as well as a set of wrapper functions that correspond to the POSIX regular expression API. The PCRE library is free, even for building commercial software."
+PKG_USE_CMAKE="no"
+PKG_AUTORECONF="no"
```
Small nit - after #2117, can you reverse the two lines above so that it's:
```
PKG_AUTORECONF="no"
PKG_USE_CMAKE="no"
```
Thanks!

thoradia commented:
@MilhouseVH I fixed according to your comment
@Raybuntu IIRC qt-everywhere requires pcre2. I rebased and fixed according to your comment.
qt-everywhere 5.9.2 is out, should I update?

Raybuntu commented:
@thoradia pcre2 provide pcre16? Look for instance here:
https://github.com/thoradia/LibreELEC.tv/blob/cb7de702fe6c0b2af4ec015657c4831dd2f75944/packages/addons/service/hyperion/patches/hyperion-0002-add-extra-linking-libraries.patch#L18
If I remember correctly hyperion.ng will require qtserialport. Maybe remove the skip?

thoradia commented:
@Raybuntu I added qtserialport. pcre and pcre2 are different major versions of the PCRE library (see https://www.pcre.org/). Using pcre2 works out of the box.
I also removed the patch, which did not apply and appears useless.
This builds now.

Raybuntu commented:
@thoradia Ahh ok missed it.

thoradia commented:
@Raybuntu Will this be merged?

Raybuntu commented:
@thoradia I will have to make a clean build to test it first. I will also include it in my builds to get some Hyperion testing. From my side there is nothing against a merge.

thoradia commented:
I merged this downstream, closing

Raybuntu commented:
??

thoradia commented:
I did not intend to maintain pull requests for weeks in addition to the corresponding code in my active branches. The code is not lost: you can find it in my branches. Feel free to pick it from there, when you have the time.

Raybuntu commented:
I'm sorry you feel that way. Sometimes it takes some time. I hope you are not discouraged to submit PR's in the future.
